### PR TITLE
Remove LCOW (step 6)

### PR DESCRIPTION
**Pull request by thaJeztah**
Splitting off more bits from https://github.com/moby/moby/pull/42170

### CODE REVIEW

**Comment by thaJeztah** [comment]
@tianon @cpuguy83 @justincormack PTAL

File: oci/defaults.go [hunk 1]
```diff
@@ -37,7 +32,7 @@ func DefaultWindowsSpec() specs.Spec {
 
 // DefaultLinuxSpec create a default spec for running Linux containers
 func DefaultLinuxSpec() specs.Spec {
-	s := specs.Spec{
+	return specs.Spec{
```
Comment (thaJeztah) [inline oci/defaults.go:35]: Changes in this file are easiest reviewed with "ignore whitespace"; https://github.com/moby/moby/pull/42683/files?w=1

**Review by AkihiroSuda — APPROVED** [review]

File: distribution/pull_v2.go [hunk 2]
```diff
@@ -487,6 +486,14 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named, platform
 }
 
 func (p *v2Puller) pullSchema1(ctx context.Context, ref reference.Reference, unverifiedManifest *schema1.SignedManifest, platform *specs.Platform) (id digest.Digest, manifestDigest digest.Digest, err error) {
+	if platform != nil {
+		// Early bath if the requested OS doesn't match that of the configuration.
```
Comment (cpuguy83) [inline distribution/pull_v2.go:490]: `Early bath`?

File: hack/ci/windows.ps1 [hunk 3]
```diff
@@ -191,7 +190,7 @@ Function Nuke-Everything {
         Stop-Process -name "tail" -Force -ErrorAction SilentlyContinue 2>&1 | Out-Null
 
         # Detach any VHDs
-        gwmi msvm_mountedstorageimage -namespace root/virtualization/v2 -ErrorAction SilentlyContinue | foreach-object {$_.DetachVirtualHardDisk() }
+        gwmi msvm_mountedstorageimage -namespace root/virtualization/v2 -ErrorAction SilentlyContinue | foreach-Object {$_.DetachVirtualHardDisk() }
```
Comment (cpuguy83) [inline hack/ci/windows.ps1:193]: Why this change?

File: hack/ci/windows.ps1 [hunk 4]
```diff
@@ -602,10 +601,10 @@ Try {
     }
 
     # Arguments: Allow setting optional storage-driver options
-    # example usage: DOCKER_STORAGE_OPTS="lcow.globalmode=false,lcow.kernel=kernel.efi"
+    # example usage: DDOCKER_STORAGE_OPTS="size=40G"
     if (-not ("$env:DOCKER_STORAGE_OPTS" -eq "")) {
         Write-Host -ForegroundColor Green "INFO: Running the daemon under test with storage-driver options ${env:DOCKER_STORAGE_OPTS}"
-        $env:DOCKER_STORAGE_OPTS.Split(",") | ForEach {
+        $env:DOCKER_STORAGE_OPTS.Split(",") | ForEach-Object {
```
Comment (cpuguy83) [inline hack/ci/windows.ps1:607]: Why the change?

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: distribution/pull_v2.go (hunk 2 above)
Comment (thaJeztah) [reply distribution/pull_v2.go:490]: Comment was already there. I can only assume John wrote that, as it looks to be an 🇬🇧 expression https://dictionary.cambridge.org/dictionary/english/an-early-bath

File: hack/ci/windows.ps1 (hunk 4 above)
Comment (thaJeztah) [reply hack/ci/windows.ps1:607]: Linter complaining that Foreach is an alias, and that should not be used as it could lead to unpredictable behavior

File: hack/ci/windows.ps1 (hunk 3 above)
Comment (thaJeztah) [reply hack/ci/windows.ps1:193]: Hmm I think that was when I fixed the alias, but I see `Foreach` does not have a capital here either; must've overlooked that when extracting this from the other PR

**Review by cpuguy83 — APPROVED** [review]
LGTM
